test/test_tagging.py

```diff
@@ -0,0 +1,110 @@
+import json
+from unittest.mock import (
+    PropertyMock,
+    patch,
+)
+
+from azul.deployment import (
+    populate_tags,
+)
+from azul.logging import (
+    configure_test_logging,
+)
+from azul_test_case import (
+    AzulUnitTestCase,
+)
+
+
+def setupModule():
+    configure_test_logging()
+
+
+class TestTagging(AzulUnitTestCase):
+
+    @patch('subprocess.run', new_callable=PropertyMock)
+    def test(self, terraform_mock):
+        # We mock Terraform since it's not available on Travis. The mocked
```

Comment:
This comment is misleading. You'd want to mock even if the `terraform` binary were available on Travis.

Reply:
Why would we want to mock even if the `terraform` binary were available on Travis?

Jesse and I discussed this a little bit, and one of the conclusions that we reached is that the schema provided by `terraform providers schema -json` is dependent on the providers used, and not the presence of individual resources, which was one of the reasons why I chose this approach.

Thinking about it now, is the reason we want to mock anyway because we can't guarantee what resources AWS/Terraform will designate as taggable?

Reply:
Unit tests should have minimal external requirements and be quick. It should be obvious to the informed reader why this is mocking the `terraform` stdout, even without the comment. The reference to Travis is misleading in that it seems to suggest that in other contexts (say, Gitlab, where `terraform` is installed) we would not want to mock.